ddl/ddl_api.go

```diff
@@ -5296,6 +5299,9 @@ func (d *ddl) DropForeignKey(ctx sessionctx.Context, ti ast.Ident, fkName model.
 	if err != nil {
 		return errors.Trace(infoschema.ErrTableNotExists.GenWithStackByArgs(ti.Schema, ti.Name))
 	}
+	if t.Meta().TempTableType != model.TempTableNone {
```

Comment:
Why check it in `DropForeignKey`? Is there any case in which we should check it?

Reply:
alter table xx drop foreign key ...

Reply:
don't we need to check it?

Reply:
Now that temporary tables won't have foreign keys, dropping foreign keys from these tables will report errors.